src/proto/beats.proto

```diff
@@ -10,22 +11,32 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum Type {
+  BEAT = 0;
+  BAR = 1;
+  PHRASE = 2;
+}
+
+message TimeSignature {
+  required int32 beats_per_bar = 1 [ default = 4];
+  required int32 note_value = 2 [ default = 4 ];
 }
 
 message Bpm {
-  optional double bpm = 1;
+  required double bpm = 1;
```

Comment:
I strongly suggest we avoid required fields -- they are broadly considered an anti-pattern in proto2 and are removed in proto3.